src/lib/mongoDriver/mongoConnectionPool.cpp

```diff
@@ -329,54 +330,76 @@ static std::string composeMongoUri
 {
   // Compose the mongoUri, taking into account all information
 
-  std::string uri = "mongodb://";
+  std::string uri;
 
-  // Add auth parameter if included
-  if (strlen(username) != 0 && strlen(passwd) != 0)
+  if (strlen(dbURI) != 0)
   {
-    uri += username + std::string(":") + passwd + "@";
+    const char* pwd = strstr(dbURI, "${PWD}");
+    if (pwd != NULL)
+    {
+      if (strlen(passwd) == 0)
+      {
+        LM_X(1, ("Invalid Command Line Options: -dbURI is used with a password substitution, but no password (-dbpwd) is supplied"));
+      }
+
+      uri = std::string(dbURI, pwd - dbURI) + passwd + (pwd + 6);
+    }
```

Comment:
Why implementing this password substitution mechanism? Why don't just use the password in the `-dbUri` parameter? Could you elaborate on this?

Reply:
This is because Orion-LD implements that functionality.

See also following comment:
https://github.com/telefonicaid/fiware-orion/issues/3794#issuecomment-1329403871